src/de/mod.rs

```diff
@@ -409,8 +409,10 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
             } else {
                 self.bytes.err(ErrorCode::ExpectedStructEnd)
             }
-        } else {
+        } else if name.is_empty() {
             self.bytes.err(ErrorCode::ExpectedStruct)
```

Comment:
Are there cases where `name` is empty?

Reply:
For newtype structs this would only be the case if a serde impl gives us an empty name. For structs it can actually happen when a struct variant is deserialised, as I did not find a way to get the name of the variant. I just added the same check here for consistency.